ext/json/tests/json_validate_005.phpt

```diff
@@ -0,0 +1,58 @@
+--TEST--
+json_validate() - Tests json_validate() performance
+--INI--
+memory_limit=128M
+--FILE--
+<?php
+
+//Generate a JSON STring
+$limit = 40000;
+$jsonString = '{ "test": { "foo": "bar" },';
+for ($i=0; $i < $limit; $i++) {
+  $jsonString .= " \"test$i\": { \"foo\": { \"test\" : { \"foo\" :  { \"test\" : { \"foo\" : \"bar\" }}}}},";
+}
+$jsonString .= ' "testXXXX": { "foo": "replaceme" } }';
+
+//Start json_decode() test
+
+$memoryBefore = memory_get_usage(true) / 1024 / 1024;
+$memoryBeforePeak = memory_get_peak_usage(true) / 1024 / 1024;
+$start = microtime(true);
+
+json_decode($jsonString, true, $limit, 0);
+
+$memoryAfter = memory_get_usage(true) / 1024 / 1024;
+$memoryAfterPeak = memory_get_peak_usage(true) / 1024 / 1024;
+
+$json_decode_memory_usage = ($memoryAfter - $memoryBefore);
+$json_decode_memory_peak_usage = ($memoryAfterPeak - $memoryBeforePeak);
+$json_decode_time = (microtime(true) - $start);
+
+unset($memoryBefore);
+unset($memoryBeforePeak);
+unset($start);
+unset($memoryAfter);
+unset($memoryAfterPeak);
+
+//Start json_validate() tests
+
+$memoryBefore = memory_get_usage(true) / 1024 / 1024;
+$memoryBeforePeak = memory_get_peak_usage(true) / 1024 / 1024;
+
+json_validate($jsonString);
+
+$memoryAfter = memory_get_usage(true) / 1024 / 1024;
+$memoryAfterPeak = memory_get_peak_usage(true) / 1024 / 1024;
+
+$json_validate_memory_usage = ($memoryAfter - $memoryBefore);
+$json_validate_memory_peak_usage = ($memoryAfterPeak - $memoryBeforePeak);
```

Comment:
I'd just get rid of this test file entirely.

If there was a memory leak, other tests would fail when run in `--enable-debug` PHP builds.

Also, memory_get_peak_usage never decreases (it's the largest amount of memory the program has **ever** used), so it isn't actually testing the right thing. This will be 0. Calling json_decode a second time of json_validate would also be `$json_validate_memory_peak_usage = (nearly) 0`

You aren't keeping a reference to the result of json_decode so the memory_get_usage(true) would just indicate memory that was given to php's internal allocator rather than the OS's memory tracker

- This test would also fail if php is compiled without support for memory usage tracking

```
ZEND_API size_t zend_memory_usage(bool real_usage)
{
#if ZEND_MM_STAT
	if (real_usage) {
		return AG(mm_heap)->real_size;
	} else {
		size_t usage = AG(mm_heap)->size;
		return usage;
	}
#endif
	return 0;
}
// zend_memory_peak_usage is similar
```
https://www.php.net/memory_get_peak_usage

Reply:
Honestly, the feedback is somehow too technical for me. 

But you identify a potential problem with it, and that is enough reason for me to take it out.

I will take it out